From 957f5fc64b577254d19d0e862cb9d6fcaf6690ec Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Wed, 20 Nov 2024 00:14:04 +0100
Subject: [PATCH] [DogeCash] `uint32_t`ify `AUXPOW_CHAIN_ID`, ` nIndex` and
 `nChainIndex` (#51)

`AUXPOW_CHAIN_ID` is only compared to `uint32_t` types, so it should be that.
`nIndex` and `nChainIndex` are unused on master, but it's clear later that they should be uint32_t too.

(cherry picked from commit 47b04d02489d0613cbd7bba260c589a1e5778b1a)

diff --git a/src/primitives/auxpow.h b/src/primitives/auxpow.h
index 7e7020e3351ccaca5cb9e6ad48cb10ce2324eebc..9a2b03d60144ce1f3726137a7338471f9edd96e3 100644
--- a/src/primitives/auxpow.h
+++ b/src/primitives/auxpow.h
@@ -18,7 +18,7 @@ static constexpr int32_t VERSION_AUXPOW_BIT = 1 << VERSION_AUXPOW_BIT_POS;
 static constexpr int32_t VERSION_CHAIN_ID_BIT_POS = 16;
 
 /** Chain ID used by Dogecoin. */
-static constexpr int32_t AUXPOW_CHAIN_ID = 0x62;
+static constexpr uint32_t AUXPOW_CHAIN_ID = 0x62;
 
 /** Max allowed chain ID */
 static constexpr uint32_t MAX_ALLOWED_CHAIN_ID =
@@ -83,13 +83,13 @@ public:
     CTransactionRef coinbaseTx;
     uint256 hashBlock;
     std::vector<uint256> vMerkleBranch;
-    int32_t nIndex;
+    uint32_t nIndex;
 
     /** The merkle branch connecting the aux block to our coinbase. */
     std::vector<uint256> vChainMerkleBranch;
 
     /** Merkle tree index of the aux block header in the coinbase. */
-    int32_t nChainIndex;
+    uint32_t nChainIndex;
 
     /** Parent block header (on which the real PoW is done). */
     CBaseBlockHeader parentBlock;
